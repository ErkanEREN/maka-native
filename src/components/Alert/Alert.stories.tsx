import React from 'react'
import { View } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import Alert from './Alert'

storiesOf('Alert', module).add('Alerts', () => (
  <View>
    <Alert message="Message" type="success" />
    <Alert message="Message" type="warning" />
    <Alert message="Message" type="error" />
  </View>
))
