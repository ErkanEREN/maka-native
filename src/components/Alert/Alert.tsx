import React from 'react'
import { View, Text } from 'react-native'
import styled from 'styled-components'

const typeMap = {
  success: 'green',
  warning: 'orange',
  error: 'red',
}

const Container = styled(View)<{ type: 'success' | 'warning' | 'error' }>`
  text-align: center;
  padding: 10px;
  background-color: ${({ type }) => typeMap[type]};
`

const AlertText = styled(Text)`
  color: #ffffff;
  text-align: center;
`

interface Props {
  message: string
  type: 'success' | 'warning' | 'error'
}

export default function Alert({ type, message }: Props) {
  return (
    <Container type={type}>
      <AlertText>{message}</AlertText>
    </Container>
  )
}
