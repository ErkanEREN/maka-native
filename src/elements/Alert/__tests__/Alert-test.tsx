import React from "react";
import { shallow } from "enzyme";
import Alert from "../Alert";

describe("Alert", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const message = "alert message";
      const component = shallow(<Alert message={message} />);
      expect(component).toMatchSnapshot();
    });
  });
});
