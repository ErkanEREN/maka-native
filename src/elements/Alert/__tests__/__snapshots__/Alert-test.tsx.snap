// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Styled(Text)>
    alert message
  </Styled(Text)>
</Styled(View)>
`;
